### PR TITLE
- Created model User and Dao layer for it;

### PR DESCRIPTION
**Pull request by NikolayFatkullin**
- Created UserService;

### CODE REVIEW

**Review by Fedich09 — APPROVED** [review]

File: src/main/java/spring/intro/dao/impl/UserDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+@Repository
+public class UserDaoImpl implements UserDao {
+    private final SessionFactory sessionFactory;
+
+    public UserDaoImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
+
+    public void add(User user) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/spring/intro/dao/impl/UserDaoImpl.java:20]: let's use Override annotation

File: src/main/java/spring/intro/dao/impl/UserDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+            }
+            throw new DataProcessingException("Can't insert User entity: " + user, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    public List<User> listUsers() {
```
Comment (dmytro-hryhoruk) [inline src/main/java/spring/intro/dao/impl/UserDaoImpl.java:40]: same

File: src/main/java/spring/intro/service/impl/UserServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+@Service
+public class UserServiceImpl implements UserService {
+    private final UserDao userDao;
+
+    public UserServiceImpl(UserDao userDao) {
+        this.userDao = userDao;
+    }
+
+    public void add(User user) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/spring/intro/service/impl/UserServiceImpl.java:17]: same

File: src/main/java/spring/intro/service/impl/UserServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        userDao.add(user);
+    }
+
+    public List<User> listUsers() {
```
Comment (dmytro-hryhoruk) [inline src/main/java/spring/intro/service/impl/UserServiceImpl.java:21]: same

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**